src/main/lombok/eu/coinform/gateway/controller/UserController.java

```diff
@@ -133,6 +111,26 @@ public LoginResponse login(HttpServletResponse response) {
         return new LoginResponse(token);
     }
 
+    private Optional<Cookie> findCookie(String key, HttpServletRequest request){
+        if(request.getCookies() == null){
+            return Optional.empty();
+        }
+        return Arrays.stream(request.getCookies())
+                .filter(cookie -> key.equals(cookie.getName()))
+                .findAny();
+    }
+
+    private String jwtTokenCreator(User user, Collection<GrantedAuthority> authorities){
+        return (new JwtToken.Builder())
+                .setSignatureAlgorithm(SignatureAlgorithm.HS512)
+                .setKey(signatureKey)
+                .setCounter(user.getCounter())
+                .setExpirationTime(2*60*1000L)
```

Comment:
That would be 2 minutes of JWT life (if I'm not wrong). As specified in David's proposal (https://github.com/co-inform/Gateway/issues/19) it should be changed to 2 hours ;)

Reply:
nice catch!